routes/unsafe-file-upload/index.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+const fs = require('fs');
+const path = require('path');
+
+exports.name = 'hapitestbench.unsafefileupload';
+
+
+exports.register = function unsafeFileUpload(server, options) {
+  server.route([
+    {
+      method: 'GET',
+      path: '/',
+      handler: {
+        view: 'unsafe-file-upload'
+      },
+    },
+    {
+      method: 'POST',
+      path: '/submit',
+      options: {
+        payload: {
+          output: 'stream',
+          parse: true,
+          allow: 'multipart/form-data'
+        }
+      },
+      handler: (request, h) => {
+        const payload = request.payload;
+
+        if (payload.file) {
+          const name = payload.file.hapi.filename;
+          const filePath = path.join(__dirname, '../../uploads', name);
```

Comment:
this should just be a path.resolve (each `'..'` as a separate arg) so that the file path doesn't look like `/app/a/b/../../uploads/blah`